src/main/java/edu/cornell/kfs/module/purap/document/validation/impl/CuAccountsPayableBankCodeValidation.java

```diff
@@ -1,55 +1,61 @@
-                isValid = CuBankCodeValidation.validate(apDocument.getBankCode(), "document." + PurapPropertyConstants.BANK_CODE, ((CuPaymentRequestDocument)apDocument).getPaymentMethodCode(), false, true);            
-                if ( isValid ) {
-                    if ( !(event instanceof AttributedRouteDocumentEvent) &&  StringUtils.isNotBlank(apDocument.getBankCode())
-                        && !CuBankCodeValidation.doesBankCodeNeedToBePopulated(((CuPaymentRequestDocument)apDocument).getPaymentMethodCode()) ) {
-                        apDocument.setBank(null);
-                        apDocument.setBankCode(null);                
-                    }
-                }
-        	}
-        }  else if ( apDocument instanceof VendorCreditMemoDocument ) {
-        	if (StringUtils.isNotBlank(apDocument.getBankCode())) {
-        	     isValid = CuBankCodeValidation.validate(apDocument.getBankCode(), "document." + PurapPropertyConstants.BANK_CODE,  ((CuVendorCreditMemoDocument)apDocument).getPaymentMethodCode(), false, true);                        
-        	     if ( isValid ) {
-                // clear out the bank code on the document if not needed (per the message set by the call above)
-        	          if ( StringUtils.isNotBlank(apDocument.getBankCode())
-        	        		        && !CuBankCodeValidation.doesBankCodeNeedToBePopulated(((CuVendorCreditMemoDocument)apDocument).getPaymentMethodCode()) ) {
-        	        	    apDocument.setBank(null);
-        	        	    apDocument.setBankCode(null);                
-        	        }
-               }
-        	}
-        } else {
-            isValid = CuBankCodeValidation.validate(apDocument.getBankCode(), "document." + PurapPropertyConstants.BANK_CODE, false, true);
+        if (!isDocumentTypeUsingBankCode(apDocument)) {
+            return true;
         }
 
+        String paymentMethodCode = getPaymentMethodCodeFromDocumentIfSupported(apDocument);
+        boolean isValid = CuBankCodeValidation.validate(
+                apDocument.getBankCode(), KFSPropertyConstants.DOCUMENT + KFSConstants.DELIMITER + PurapPropertyConstants.BANK_CODE,
+                paymentMethodCode, false, true);
+
         return isValid;
-	}
+    }
+
+    // This method is private on the superclass, so it has been copied into this class and tweaked accordingly.
+    protected boolean isDocumentTypeUsingBankCode(AccountsPayableDocumentBase apDocument) {
+        String documentTypeName = apDocument.getDocumentHeader().getWorkflowDocument().getDocumentTypeName();
+        ParameterEvaluator evaluator = getParameterEvaluatorService().getParameterEvaluator(
+                Bank.class, KFSParameterKeyConstants.BANK_CODE_DOCUMENT_TYPES, documentTypeName);
+        return evaluator.evaluationSucceeds();
+    }
+
+    protected ParameterEvaluatorService getParameterEvaluatorService() {
+        return SpringContext.getBean(ParameterEvaluatorService.class);
```

Comment:
should ParameterEvaluatorService be made a member variable, and only call sprincontext here if the member variable isn't set?

Reply:
Yes, I think I can refactor this to use Spring to inject the service. Thanks for the suggestion.